pkg/alertmanager/multitenant.go

```diff
@@ -737,6 +774,24 @@ func (am *MultitenantAlertmanager) ServeHTTP(w http.ResponseWriter, req *http.Re
 		return
 	}
 
+	if am.cfg.ShardingEnabled && am.distributor.IsPathSupported(req.URL.Path) {
+		am.distributor.DistributeRequest(w, req)
+		return
+	}
+
+	// If sharding is not enabled or Distributor does not support this path,
+	// it is served by this instance.
+	am.serveHTTPHelper(w, req)
+}
+
+// HandleRequest serves the Alertmanager's web UI and API sent via gRPC.
```

Comment:
I'd suggest extending this comment a bit:

```suggestion
// HandleRequest implements gRPC Alertmanager service, which receives request from AlertManager-Distributor
// and forwards them to internal http handler (without doing the distribution logic again).
```